src/gpu/intel/jit/v2/conv/planner/bench.cpp

```diff
@@ -614,6 +614,7 @@ bench_data_t bench(const bench_manager_t &bench_mger,
 
 bool try_create(
         const bench_manager_t &bench_mger, const kernel_desc_t &kernel_desc) {
+    clear_primitive_cache();
```

Comment:
Would it be reasonable to just set the primitive cache capacity to 0?

Reply:
We still want to enable kernel cache to reduce primitive creation time in general. In a typical benchmark run we create 100+ primitives with different shapes but all of them use the same kernel underneath - kernel/primitive cache helps with that.

A caveat here is that shapes may repeat, and we want to test the same shape with a different kernel. In such places we need to clear the cache to avoid undesired primitive cache hits. However we still need primitive cache enabled for other cases.

Reply:
This is not working as you intend. The external [primtive_cache interace](https://github.com/oneapi-src/oneDNN/blob/6bc11a06251a230b8fd2c3ab8b355c37e881045e/src/common/primitive_cache.cpp#L150-L157) (called from [here](https://github.com/oneapi-src/oneDNN/blob/6bc11a06251a230b8fd2c3ab8b355c37e881045e/src/gpu/intel/jit/v2/conv/planner/bench.cpp#L518))  will also clear the kernel cache because the kernel_cache is not intended to be externally visible. You will need to add a new interface or environment variable to get your desired behavior.

Reply:
Got you now. So for your suggestion we need to introduce an interface to control kernel/primitive caches separately and use it instead (now these abstractions are hidden even internally).

I implemented this in the last commit, please take a look.

Also tagging @densamoilov to review primitive cache changes.